gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -118,7 +123,13 @@ QueryRequest buildQueryRequest(
               .orElse(request.getFilter());
     }
 
+    if (entityIds.isEmpty()) {
```

Comment:
entityIds starts empty and only gets filled if the condition on 119 is met - so that means that this would always short-circuit if we have a metric source, right?

Reply:
good catch. moved the empty entity ids check inside the attribute source condition